### ISSUE TITLE
Feature request: Support for Microsoft's new MSIX installer format

### ISSUE THREAD

**Issue by jakubsuchybio** [wixtoolset/issues#5781]
* Describe the scenario and benefits that the feature supports.
Microsoft released new MSIX installer format.
More info here: https://www.reddit.com/r/Windows10/comments/837saq/microsoft_announces_msix_installer_format_that/
Will you support it in your WixToolset in the future?
It looks really promissing.

**Comment by robmen** [comment]
As we understand it at this time, MSIX is essentially a rename of Desktop Bridge use of AppX. FireGiant already provides tooling that supports creating [AppX using from WiX Toolset][appx].

[appx]: https://firegiant.com/r/appx

**Comment by jakubsuchybio** [comment]
Well that is not my understanding of it.
I think it is more like having win32 app with msi installer and making it msix that can also be submitted into Windows Store. But can be also installed manually on older system without Windows Store.
I might be wrong though.

**Comment by robmen** [comment]
Wow, I've not heard of that capability at all. But there hasn't been a lot of solid technical details made available. I was told we'd get more details in a couple months or so.

**Comment by jakubsuchybio** [comment]
Yes :) well it is still in diapers. But if they can deliver, this could be huge. I just think that you would want to support that in wix.

**Comment by robmen** [comment]
I know FireGiant is very much down that path.

**Comment by rogersachan** [comment]
Microsoft is creating a new, open-source packaging format that is not locked down to their store. This format has support for both Windows 7 and 10. They are working with multiple partners for this packaging as seen below: 
![image](https://user-images.githubusercontent.com/7173984/39970947-966af08e-56c1-11e8-80e4-09697950b59b.png)

See:

* [MSIX: Inside and Out](https://youtu.be/FKCX4Rzfysk)
* [Accelerating Windows 10 enterprise app deployment with MSIX](https://youtu.be/z6q15jwk0H8)
* [MSIX - Microsoft Tech Community](https://techcommunity.microsoft.com/t5/MSIX/ct-p/MSIX)
* [MSIX on Github](https://github.com/Microsoft/msix-packaging)

@robmen I think this packaging format has a larger future than the existing bridge with Win7 support. With all those partners with them this may be worth investigating.

**Comment by jozefizso** [comment]
Is MSIX really supported on Windows 7? I though it is exlusive to the latest Windows 10 releases.

**Comment by rogersachan** [comment]
@jozefizso See the first video at this timestamp: https://youtu.be/FKCX4Rzfysk?t=2947

**Comment by jozefizso** [comment]
That's a prototype - a custom script which just unzips the files. They explicitly say that none of the MSIX features are supported on Windows 7.

**Comment by BMurri** [comment]
The repo is just a "subset of the functionality contained within appxpackaging.dll on Windows" without any instruction on MSIX layouts/differences from AppX, etc. So... FireGiant currently has as much working code for this as anyone else possibly could. As soon as there's some actionable information released, we'll have something we can work on.

The WIP would then be based on that actionable info, and any tasks needed would be assigned out once the path forward is agreed to.

I agree with you on their abandonware problem.

**Comment by barnson** [comment]
Also worth noting: http://www.joyofsetup.com/2018/05/19/msix-slightly-more-detail-about-what-might-be-something-huge-but-probably-wont-be/

**Comment by RaulRG** [comment]
We develop a client-server software and use WiX for installing a Windows service. The clients for this Windows service are deployed using ClickOnce. We would love to use MSIX for this as ClickOnce seems to be on a roadblock. A +1 from us to this request!

**Comment by superusercode** [comment]
Looks like MSIX support for WiX is a paid offering managed by FireGiant now https://www.firegiant.com/products/wix-expansion-pack/msix/

If FireGiant does it on their own does it mean WiX will never have it (i.e. how the gitlab CE/EE model works)?

**Comment by rseanhall** [comment]
No. What FireGiant implements has no bearing on what we will accept into the WiX Toolset. If we weren't going to allow the feature then we would have closed this issue.

That being said, this is a massive feature. As the wip-required label indicates, it will require a WiX Improvement Proposal to discuss the design of the feature to make sure it fits in with the rest of the toolset. "Drive-by" pull requests would not be accepted. As with any large feature, part of the decision of accepting it would be figuring out who's going to maintain it. If the core team has doubts about whether the implementer will stick around and provide ongoing maintenance, and no one on the core team is interested in maintaining it, then that would be a problem.

**Comment by madaster97** [comment]
Hi there,

I've been researching this topic to access some UWP features in non-UWP apps. Wanted to call out these 2 Microsoft articles that came out this summer:
[MSIX Packaging Tool](https://docs.microsoft.com/en-us/windows/msix/packaging-tool/tool-overview)
[Building MSIX from source code (without UWP)](https://docs.microsoft.com/en-us/windows/msix/desktop/source-code-overview)

I'm not sure if/how these could be integrated with Wix, but for a start the second article seems to enable UWP features for non-UWP apps.

**Comment by robmen** [comment]
FireGiant provides a [WiX Extension that builds MSIX packages](https://www.firegiant.com/products/wix-expansion-pack/msix/). Note that is a commercial solution today because it took significant resources to develop.

**Comment by tjmoore** [comment]
MSIX appears to have the advantage of being able to build on Linux?

Very useful for those with Linux based CI. Build anywhere.
